Decisions.Equifax/EquifaxUtilities.cs

```diff
@@ -0,0 +1,159 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Text.Encodings.Web;
+using Decisions.Equifax.ConsumerCreditReport.Request;
+using Decisions.Equifax.ConsumerCreditReport.Response;
+using Decisions.Equifax.PreQualificationOfOne.Response;
+using DecisionsFramework;
+using DecisionsFramework.Design.ConfigurationStorage.Attributes;
+using DecisionsFramework.Design.Flow;
+using DecisionsFramework.Design.Flow.CoreSteps;
+using DecisionsFramework.Design.Flow.Mapping;
+using DecisionsFramework.Design.Properties.Attributes;
+using DecisionsFramework.ServiceLayer;
+using DecisionsFramework.ServiceLayer.Services.ContextData;
+using Newtonsoft.Json;
+namespace Decisions.Equifax
+{
+    public class EquifaxUtilities
+    {
+        public static readonly Log log = new Log(EquifaxConstants.LogCat);
+
+        internal static ConsumerCreditReportResponse ExecuteCreditReportRequest(ConsumerCreditReportRequest request, string scope, string requestUrl)
+        {
+            string responseString = RequestSerializer(request, scope, requestUrl);
+
+            ConsumerCreditReportResponse limitedCreditResponse = new ConsumerCreditReportResponse();
+           
+                limitedCreditResponse = JsonConvert.DeserializeObject<ConsumerCreditReportResponse>(
+                    responseString, new JsonSerializerSettings {NullValueHandling = NullValueHandling.Ignore});
+                return limitedCreditResponse;
+        }
+        internal static PreQualificationOfOneResponse ExecutePrequalificationRequest(ConsumerCreditReportRequest request, string scope, string requestUrl)
+        {
+            string responseString = RequestSerializer(request, scope, requestUrl);
+            PreQualificationOfOneResponse preQualResponse = new PreQualificationOfOneResponse();
+            preQualResponse  = JsonConvert.DeserializeObject<PreQualificationOfOneResponse>(
+                responseString, new JsonSerializerSettings {NullValueHandling = NullValueHandling.Ignore});
+            
+            return preQualResponse;
+        }
+
+        private static string  RequestSerializer(ConsumerCreditReportRequest request, string scope, string requestUrl)
+        {
+            JsonSerializerSettings jsonSettings = new JsonSerializerSettings
+            {
+                NullValueHandling = NullValueHandling.Ignore
+            };
+
+            if (string.IsNullOrWhiteSpace(scope))
+                throw new LoggedException(EquifaxConstants.SETTINGS_CONFIGURATION_EXCEPTION);
```

Comment:
Found the exception we're throwing for scope missing. We should move this to the step itself, just so we can handle it out in the open where we typically expect to do it. 

This works, but, we're in RequestSerializer() so we've mostly lost the context of where this value comes from.

Generally for this method signature, we should be marking the parameters as nullable if we intend to sometimes provide them with 'null'. Baring that, we should be checking for valid values before calling the method.